### ISSUE TITLE
Parameter support for cp.convolve missing

### ISSUE THREAD

**Issue by phschiele** [cvxpy/cvxpy#2218]
**Describe the bug**
`cp.convolve(a, x)` can give wrong results when `a` is parametrized. Parametrized `x` (and constant `a`) appears to be correct.

**To Reproduce**
```py
n = 3
x = cp.Variable(n)
p = cp.Parameter(n)

problem = cp.Problem(cp.Minimize(cp.sum(cp.convolve(p, x))), [0 <= x, x<= 1])
assert problem.is_dpp()

p.value = np.arange(n)
problem.solve(canon_backend=cp.CPP_CANON_BACKEND)
>>> 0  # correct

p.value = -np.arange(n)
problem.solve(canon_backend=cp.CPP_CANON_BACKEND)
>>> 0  # incorrect

problem = cp.Problem(cp.Minimize(cp.sum(cp.convolve(p, x))), [x == 1])
problem.solve(canon_backend=cp.CPP_CANON_BACKEND)
>>> -9  # correct
```

**Expected behavior**
Solve correctly or have `problem.is_dpp()` return `False`.

**Version**
 - OS: Ubuntu
 - CVXPY Version: master (586ef7c)

**Additional context**
- Similar issue before with `cp.kron`: https://github.com/cvxpy/cvxpy/issues/1656 and links therein
- Python backends raise `AssertionError`, so only C++ is affected

**Comment by chofchof** [comment]
Even for 1-dim case, `cp.convolve(p, x)` does not give a correct answer if the parameter `p` is negative.
```
import cvxpy as cp

x = cp.Variable()
p = cp.Parameter()
problem = cp.Problem(cp.Minimize(cp.convolve(p, x)), [0 <= x, x <= 1])

p.value = -1.0
problem.solve(canon_backend=cp.CPP_CANON_BACKEND)
>>> -1.274313644086471e-09  # 0, not -1
```

I hope to see some warning message in the documentation.

**Comment by chofchof** [comment]
For 2-dim case, cp.convolve(p, x) does not give a correct answer even if the parameter p is nonnegative.
```
import cvxpy as cp

x = cp.Variable(2)
p = cp.Parameter(2)
problem = cp.Problem(cp.Minimize(cp.convolve(p, x)[1]), [0 <= x, x <= 1])

p.value = [0.0, 1.0]
problem.solve(canon_backend=cp.CPP_CANON_BACKEND)
>>> 0.4999999826934395  # 0.5, not 0 when x = [0, 0]
```

**Comment by SteveDiamond** [comment]
@chofchof can you try your example with this PR? https://github.com/cvxpy/cvxpy/pull/2427

**Comment by chofchof** [comment]
> @chofchof can you try your example with this PR? #2427

Sure. I could see the correct answer with your PR. Good work.
```
>>> 4.098987853038662e-11
```

**Comment by SteveDiamond** [comment]
Closed by #2427
